### PR TITLE
Fix the man/Dockerfile for arm

### LINKED ISSUE #25124: make deb on ARM is broken in 1.12.0-rc5
With docker 1.12.0-rc5 running a `make deb` on ARM is broken as it tries to pull and run the `golang:1.6.3-alpine` image for amd64 generating the manpages.

**Output of `docker version`:**

```
Client:
 Version:      1.10.0
 API version:  1.22
 Go version:   go1.4.3
 Git commit:   590d5108
 Built:        Thu Feb  4 21:04:51 2016
 OS/Arch:      linux/arm

Server:
 Version:      1.10.0
 API version:  1.22
 Go version:   go1.4.3
 Git commit:   590d5108
 Built:        Thu Feb  4 21:04:51 2016
 OS/Arch:      linux/arm
```

**Output of `docker info`:**

```
Containers: 22
 Running: 1
 Paused: 0
 Stopped: 21
Images: 363
Server Version: 1.10.0
Storage Driver: overlay
 Backing Filesystem: extfs
Execution Driver: native-0.2
Logging Driver: json-file
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 4.5.2-std-2
Operating System: Ubuntu 16.04 LTS
OSType: linux
Architecture: armv7l
CPUs: 4
Total Memory: 1.975 GiB
Name: jenkins-slave01
ID: DFJ3:7BSF:SOGE:A5CN:HZT3:3Y35:UITQ:EEH7:YDHI:FFLX:RUXV:EBUA
WARNING: No cpu cfs quota support
WARNING: No cpu cfs period support
Labels:
 provider=scaleway
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Scaleway C1 with Docker preinstalled

**Steps to reproduce the issue:**
1. Use eg. a Scaleway C1 ARM machine
2. git clone https://github.com/docker/docker
3. cd docker
4. make deb

**Describe the results you received:**

```
---> Making bundle: dynbinary (in bundles/1.12.0-rc5/dynbinary)
Building: bundles/1.12.0-rc5/dynbinary-client/docker-1.12.0-rc5
Created binary: bundles/1.12.0-rc5/dynbinary-client/docker-1.12.0-rc5
Building: bundles/1.12.0-rc5/dynbinary-daemon/dockerd-1.12.0-rc5
Created binary: bundles/1.12.0-rc5/dynbinary-daemon/dockerd-1.12.0-rc5
Building: bundles/1.12.0-rc5/dynbinary-daemon/docker-proxy-1.12.0-rc5
Created binary: bundles/1.12.0-rc5/dynbinary-daemon/docker-proxy-1.12.0-rc5

---> Making bundle: build-deb (in bundles/1.12.0-rc5/build-deb)
---> Making bundle: .integration-daemon-start (in bundles/1.12.0-rc5/build-deb)
INFO: Waiting for daemon to start...
+++ exec dockerd --debug --host unix:///go/src/github.com/docker/docker/bundles/1.12.0-rc5/build-deb/docker.sock --storage-driver overlay --pidfile bundles/1.12.0-rc5/build-deb/docker.pid --userland-proxy=true
.
---> Making bundle: .detect-daemon-osarch (in bundles/1.12.0-rc5/build-deb)
docker build -t docker-manpage-dev -f man/Dockerfile .
Sending build context to Docker daemon 557.1 kB
...
Sending build context to Docker daemon 157.5 MB

Step 1 : FROM golang:1.6.3-alpine
1.6.3-alpine: Pulling from library/golang
e110a4a17941: Pulling fs layer
...
25994cc11974: Pull complete
Digest: sha256:8b9e1e4a137e7663c1dc52a33c41699661ea4da9aca6ab0771b1fcec16a87535
Status: Downloaded newer image for golang:1.6.3-alpine
 ---> 6b317e0ae72c
Step 2 : RUN apk add -U git bash curl gcc musl-dev
 ---> Running in 23bd447afd78
standard_init_linux.go:175: exec user process caused "exec format error"
The command '/bin/sh -c apk add -U git bash curl gcc musl-dev' returned a non-zero code: 1
Makefile:119: recipe for target 'manpages' failed
make: *** [manpages] Error 1
---> Making bundle: .integration-daemon-stop (in bundles/1.12.0-rc5/build-deb)
+++++ cat bundles/1.12.0-rc5/build-deb/docker.pid
++++ kill 5433
Makefile:86: recipe for target 'deb' failed
make: *** [deb] Error 1
```

**Describe the results you expected:**

Building the ARM deb's should work without a problem :-)

I had a look at [man/Dockerfile](https://github.com/docker/docker/blob/master/man/Dockerfile) to see if we can add eg. a arch specific `man/Dockerfile.armhf` but found out that glide does not provide a precompiled ARM binary. So there is some more work to make it work.

Another idea is to generate the man pages and check them into GitHub with a `[ci skip]` commit message.

### PR DESCRIPTION
**Pull request by dnephin**
Fixes #25124

cc @tiborvass

### CODE REVIEW

**Comment by justincormack** [comment]
The problem here is that each architecture needs a new Dockerfile to build the man pages.

We have an open PR for Power debs for example https://github.com/docker/docker/pull/23438 and this will now need to add a man page builder, and there is not yet a `ppc64le/alpine`, so it will have to be custom.

**Comment by justincormack** [comment]
cc @tophj-ibm

**Comment by StefanScherer** [comment]
@dnephin Thanks! I'll try that soon.
@justincormack That's right, but the previous build with `make manpages` did only work for amd64.

**Comment by dnephin** [comment]
@justincormack I understand the problem, but what is the solution? Is there some way to make a single Dockerfile work for multiple architectures?

**Comment by justincormack** [comment]
@dnephin can we add the prerequisites to build the man pages to the main Dockerfile? We already have one for each architecture there, and they have Go.

Otherwise I think vendoring is the best solution.

**Comment by dnephin** [comment]
I'm not really a fan of either of those options.

The current Dockerfile is already way too big. Putting every project task into a single image makes build tasks slow, and is hard to maintain. You can't really track which dependencies are used for which tasks. I really think we should be splitting the main Dockerfile into different ones for: build, linting, cross-build, docs (already done), and man pages.

I really dislike checking in generated code into the repo.

**Comment by StefanScherer** [comment]
@dnephin Your PR works so far. Still building the three debs for ARM, but making manpages worked fine.

**Comment by DieterReuter** [comment]
Probable in the future we could connect to a multi-arch Docker Swarm as build environment, so we could build the man pages in an amd64 container and the binaries natively on amd64/arm/ppc64le, and so on.

File: man/Dockerfile [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-RUN     export GLIDE=0.10.2; \
-        export SRC=https://github.com/Masterminds/glide/releases/download/; \
-        curl -sL ${SRC}/${GLIDE}/glide-${GLIDE}-linux-amd64.tar.gz | \
-        tar -xz linux-amd64/glide && \
-        mv linux-amd64/glide /usr/bin/glide && \
+RUN     mkdir -p /go/src /go/bin /go/pkg
+ENV     GOPATH=/go
+RUN     export GLIDE=v0.11.1; \
+        export SRC=https://dl.bintray.com/dnephin/glide/; \
+        curl -sL -o /usr/bin/glide ${SRC}/glide-${GLIDE}-linux-amd64 && \
```
Comment (justincormack) [inline man/Dockerfile]: I really don't like downloading binaries like this. There should be autobuild or official containers with the prerequisites, with a clear way for people to build them for different architectures from source if we are going to have per architecture Dockerfiles.

Comment (dnephin) [inline man/Dockerfile]: There is no official build for arm yet (it's being worked on https://github.com/Masterminds/glide/pull/528).

I can setup circle CI to build binaries from a fork I guess, but I wanted to get this fixed asap to unblock building the arm debs.  I'll look at getting that setup next week.

Comment (justincormack) [inline man/Dockerfile]: Ok, but we are talking about getting an upstream project to generate builds for eventually ppc64le, s390x, arm64, arm, and more... in order to build absolutely identical man pages. It really makes vendoring seem like a better option.

Comment (dnephin) [inline man/Dockerfile]: Only having to build the man pages on a single platform would be great. I believe we could do that with a jenkins pipeline and build artifacts.

Comment (tophj-ibm) [inline man/Dockerfile]: What about cloning and building from source like we do in the main Dockerfiles? I just tested on power and was able to make a glide binary and make the man pages from there.

**Comment by jim3mar** [comment]
@dnephin 
"I really think we should be splitting the main Dockerfile into different ones for: build, linting, cross-build, docs (already done), and man pages."
+1
Sometimes we need checkout source code and build binary, the building base image is like golang:1.6-alpine, but, the output binary just depends on alpine:xxx.

File: man/Dockerfile (hunk 1 above)
Comment (dnephin) [inline man/Dockerfile]: Building from source should work. I'll make that change

**Comment by dnephin** [comment]
Ok, glide is now built from source instead of pulled from my bintray

**Comment by StefanScherer** [comment]
I'm testing the updated PR on Scaleway

**Comment by StefanScherer** [comment]
make deb works fine on ARM with this PR

**Comment by tiborvass** [comment]
LGTM tested this too.

**Pull request merged**